src/platform_impl/linux/x11/event_processor.rs

```diff
@@ -544,74 +548,6 @@ impl<T: 'static> EventProcessor<T> {
                 }
             }
 
-            ffi::KeyPress | ffi::KeyRelease => {
-                use crate::event::ElementState::{Pressed, Released};
-
-                // Note that in compose/pre-edit sequences, this will always be Released.
-                let state = if xev.get_type() == ffi::KeyPress {
-                    Pressed
-                } else {
-                    Released
-                };
-
-                let xkev: &mut ffi::XKeyEvent = xev.as_mut();
-
-                let window = xkev.window;
-                let window_id = mkwid(window);
-
-                // Standard virtual core keyboard ID. XInput2 needs to be used to get a reliable
-                // value, though this should only be an issue under multiseat configurations.
-                let device = util::VIRTUAL_CORE_KEYBOARD;
-                let device_id = mkdid(device);
-                let keycode = xkev.keycode;
-
-                // When a compose sequence or IME pre-edit is finished, it ends in a KeyPress with
-                // a keycode of 0.
-                if keycode != 0 {
-                    let scancode = keycode - KEYCODE_OFFSET as u32;
-                    let keysym = wt.xconn.lookup_keysym(xkev);
-                    let virtual_keycode = events::keysym_to_element(keysym as c_uint);
-
-                    update_modifiers!(
-                        ModifiersState::from_x11_mask(xkev.state),
-                        self.mod_keymap.get_modifier(xkev.keycode as ffi::KeyCode)
-                    );
-
-                    let modifiers = self.device_mod_state.modifiers();
-
-                    #[allow(deprecated)]
-                    callback(Event::WindowEvent {
-                        window_id,
-                        event: WindowEvent::KeyboardInput {
-                            device_id,
-                            input: KeyboardInput {
-                                state,
-                                scancode,
-                                virtual_keycode,
-                                modifiers,
-                            },
-                            is_synthetic: false,
-                        },
-                    });
-                }
-
-                if state == Pressed {
-                    let written = if let Some(ic) = wt.ime.borrow().get_context(window) {
-                        wt.xconn.lookup_utf8(ic, xkev)
```

Comment:
I couldn't find the place to call this logic in this new change.
To get IME text, you need to call this method, and then submit `ReceivedImeText` to work with IME.

Reply:
I tried to reimplement this logic to enable IME event, but looks like IME doesn't work at all for now.
It seems that something wrong in the initialize logic.

Reply:
Can you try again with my latest changes? I think the bit you must have missed was [this event mask](https://github.com/rust-windowing/winit/pull/1932/commits/7b5a30997ccf7ca53923c5eaebcc60eaad0b7ca8#diff-7eb9227ac71734e693b64fd604d60cb5374e2601745bf4e5ae4ef619d9ef37caR196).

Reply:
Confirmed that latest version worked correctly in my x11 env.
IME worked and ReceivedImeText was submitted. Nice!

Reply:
That's great!

Reply:
I just discovered that my implementation was a tad wrong, and would emit `ReceivedImeText` for *every* key press.
Could you test that the latest changes work correctly for you?

Reply:
Sorry for super late response. Confirmed that it worked in my local environment. I got ReceivedImeText with string instead of ReceivedCharacter for every key press.

Reply:
Ok. I'll mark this as resolved then.